backend/faq/models.py

```diff
@@ -28,6 +30,55 @@ class Meta:
     def __str__(self) -> str:
         return self.name
 
+    def get_text(self, related="locales", lang=None):
+        """
+        Return the translated text of a related instance.
+
+        A related instance can be a question or an answer.
+        """
+        if lang is None:
+            lang = translation.get_language()
+        try:
+            locale = getattr(self, related).get(language=lang)
+        except ObjectDoesNotExist:
+            try:
+                locale = getattr(self, related).get(language="en")
+            except ObjectDoesNotExist:
+                return self.name  # pylint: disable=no-member
+        return locale.text
+
+    def get_text_prefetch(self, related="locales", lang=None):
+        """
+        Return the translated text of a related instance.
+
+        Contrary to `self.get_text` this method consider the related instances
+        as already prefetched with `prefetch_related`, and use `.all()` instead
+        of `.get()` to avoid triggering any additional SQL query.
+        """
+        if lang is None:
+            lang = translation.get_language()
+
+        try:
+            locale = [loc for loc in getattr(self, related).all()
+                      if loc.language == lang][0]
```

Comment:
Would this work?

```
locale = getattr(self, related).get(language=lang)
```

Reply:
I guess it would raise an error when there are several, but the unique together property is set, so it should always work?

Reply:
Yes using `.get` will work but will trigger an extra SQL query. Using `.all` allows to take advantage of an eventual call to `.prefetch_related`, which is what we use in the view.